dao/src/main/java/by/it/novik/dao/PaymentDao.java

```diff
@@ -119,6 +123,10 @@ private Criteria getCriteriaOfFilter(PaymentsFilter paymentsFilter) {
         double maxAmountPayment = paymentsFilter.getMaxAmountPayment();
         Criteria criteria = getSession().createCriteria(Payment.class);
         //To check existing filters
+        // please refactor, this is really hard to understand what us going on here and it would hard to maintain such code
+        // and moreover some conditions would literally never happen, they are unreachable code (thank to my Intellij IDEA)
+        // http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
+        // what about using Restrictions.Conjunction() method?
```

Comment:
Yes, I understand.
But you can greatly simplify your code and make it more readable if you use Restrictions.conjunction.

that's what I got

```java
private Criterion getRestriction(Date payDate, double minAmountPayment, double maxAmountPayment) {
        List<Criterion> criterions = new ArrayList<>();
        if (payDate != null) {
            criterions.add(Restrictions.eq("payDate", payDate));
        }
        if (minAmountPayment != 0 || maxAmountPayment != 0) {
            criterions.add(getAmountRestriction(minAmountPayment, maxAmountPayment));
        }
        return Restrictions.conjunction(criterions.toArray(new Criterion[criterions.size()]));
    }

    private Criterion getAmountRestriction(double minAmountPayment, double maxAmountPayment) {
        if (minAmountPayment != 0) {
            if (maxAmountPayment != 0) {
                return Restrictions.between("amountPayment", minAmountPayment, maxAmountPayment);
            }
            return Restrictions.ge("amountPayment", minAmountPayment);
        }
        return Restrictions.le("amountPayment", minAmountPayment);
    }
```